api/pkg/transformer/feast/transformer.go

```diff
@@ -233,6 +224,53 @@ func (t *Transformer) getFeastFeature(ctx context.Context, request []byte, confi
 	}, nil
 }
 
+func buildEntitiesRequest(request []byte, configEntities []*transformer.Entity) ([]feast.Row, error) {
+	allEntityRows := [][]feast.Row{}
+	for _, entity := range configEntities {
+		vals, err := getValuesFromJSONPayload(request, entity)
+		if err != nil {
+			return nil, fmt.Errorf("unable to extract entity %s: %v", entity.Name, err)
+		}
+
+		rows := []feast.Row{}
+		for _, val := range vals {
+			rows = append(rows, feast.Row{
+				entity.Name: val,
+			})
+		}
+		allEntityRows = append(allEntityRows, rows)
+	}
+
+	if len(allEntityRows) == 0 {
+		return nil, fmt.Errorf("no entity extracted")
+	}
+
+	entities := allEntityRows[0]
+
+	entityIndex := 1
+	for entityIndex < len(configEntities) {
```

Comment:
I think you can simplify this method by removing the second loop and include it in the first loop. It help avoiding too many tight loop.

Reply:
Yup, you're right. I've simplified this logic to only have one loop.